src/main/java/tdl/s3/sync/Destination.java

```diff
@@ -41,10 +57,67 @@ public AWSSecretsProvider getSecret() {
         return secret;
     }
 
-    public AmazonS3 buildClient() {
-        return AmazonS3ClientBuilder.standard()
+    public AmazonS3 getClient() {
```

Comment:
In the https://github.com/julianghionoiu/s3-sync-stream/pull/18/commits/f1ffa69968de843d8e4b75124dfe8920f9ccdf02 the `getBucket`, `getPrefix`, and `getClient` are removed.

I guess this is the best refactoring I can get without making `Destination` class really overbloated.